SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Utils/ErrorLogHelper.cs

```diff
@@ -0,0 +1,246 @@
+        /// </summary>
+        public static IEnumerable<FileInfo> GetErrorLogFiles()
+        {
+            lock (LockObject)
+            {
+                try
+                {
+                    // Convert to list so enumeration does not occur outside the lock.
+                    return CrashesDirectory.EnumerateFiles($"*{ErrorLogFileExtension}").ToList();
+                }
+                catch (System.Exception ex)
+                {
+                    AppCenterLog.Error(Crashes.LogTag, "Failed to retrieve error log files.", ex);
+                }
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets the most recently modified error log file.
+        /// </summary>
+        /// <returns>The most recently modified error log file.</returns>
+        public static FileInfo GetLastErrorLogFile()
+        {
+            FileInfo lastErrorLogFile = null;
+            lock (LockObject)
+            {
+                var errorLogFiles = GetErrorLogFiles();
+                if (errorLogFiles == null)
+                {
+                    return null;
+                }
+                try
+                {
+                    foreach (var errorLogFile in errorLogFiles)
+                    {
+                        if (lastErrorLogFile == null || lastErrorLogFile.LastWriteTime < errorLogFile.LastWriteTime)
+                        {
+                            lastErrorLogFile = errorLogFile;
+                        }
+                    }
+                    return lastErrorLogFile;
+                }
+                catch (System.Exception e)
+                {
+                    AppCenterLog.Error(Crashes.LogTag, "Encountered an unexpected error while retrieving the latest error log.", e);
+                }
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the error log file with the given ID.
+        /// </summary>
+        /// <param name="errorId">The ID for the error log.</param>
+        /// <returns>The error log file or null if it is not found.</returns>
+        public static FileInfo GetStoredErrorLogFile(Guid errorId)
+        {
+            return GetStoredFile(errorId, ErrorLogFileExtension);
+        }
+
+        /// <summary>
+        /// Saves an error log on disk.
+        /// </summary>
+        /// <param name="errorLog">The error log.</param>
+        public static void SaveErrorLogFile(ManagedErrorLog errorLog)
+        {
+            var errorLogString = LogSerializer.Serialize(errorLog);
+            var fileName = errorLog.Id + ErrorLogFileExtension;
+            try
+            {
+                lock (LockObject)
+                {
+                    if (!CrashesDirectory.Exists)
+                    {
+                        CrashesDirectory.Create();
+                    }
+                    var filePath = Path.Combine(CrashesDirectory.FullName, fileName);
```

Comment:
L160 and L161 doesn't need to be in the lock block.

Reply:
Is your concern to store same error group ID?

Reply:
The concern is that the file can be deleted somehow - but I guess that is not a big deal because we would just catch the exception. I can move it.